homeassistant/components/sonos/media_player.py

```diff
@@ -505,6 +507,25 @@ def play_media(  # noqa: C901
         If media_type is "playlist", media_id should be a Sonos
         Playlist name.  Otherwise, media_id should be a URI.
         """
+        if kwargs.get(ATTR_MEDIA_ANNOUNCE):
+            volume = kwargs.get("extra", {}).get("volume")
+            _LOGGER.debug("Playing %s using websocket audioclip", media_id)
+            try:
+                assert self.speaker.websocket
+                response, _ = run_coroutine_threadsafe(
+                    self.speaker.websocket.play_clip(
+                        media_id,
+                        volume=volume,
+                    ),
+                    self.hass.loop,
+                ).result()
```

Comment:
Maybe convert `play_media` to a coro and than run the fallback in the executor instead so we don't have to jump back and forth

Reply:
Did this partially since there are quite a few sync calls in the original method.

Reply:
Makes sense 👍